pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java

```diff
@@ -55,18 +53,16 @@ public class PulsarChannelInitializer extends ChannelInitializer<SocketChannel>
     @Getter
     private final boolean tlsEnabled;
     private final boolean tlsHostnameVerificationEnabled;
-    private final boolean tlsEnabledWithKeyStore;
     private final InetSocketAddress socks5ProxyAddress;
     private final String socks5ProxyUsername;
     private final String socks5ProxyPassword;
 
-    private final Supplier<SslContext> sslContextSupplier;
-    private NettySSLContextAutoRefreshBuilder nettySSLContextAutoRefreshBuilder;
+    private final PulsarSslFactory pulsarSslFactory;
 
     private static final long TLS_CERTIFICATE_CACHE_MILLIS = TimeUnit.MINUTES.toMillis(1);
 
-    public PulsarChannelInitializer(ClientConfigurationData conf, Supplier<ClientCnx> clientCnxSupplier)
-            throws Exception {
+    public PulsarChannelInitializer(ClientConfigurationData conf, Supplier<ClientCnx> clientCnxSupplier,
+                                    ScheduledExecutorProvider scheduledExecutorProvider) throws Exception {
```

Comment:
It seems that this could simply take a ScheduledExecutorService. The coupling to ScheduledExecutorProvider seems unnecessary.